### PR TITLE
check the combination of index type and data type

### PR DESCRIPTION
**Pull request by cqy123456**

### CODE REVIEW

**Review by liliu-z — APPROVED** [review]

File: include/knowhere/comp/knowhere_check.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            return index_factory.HasIndex<bin1>(index_name);
+        case VecType::VECTOR_FLOAT:
+            return index_factory.HasIndex<fp32>(index_name);
+        case VecType::VECTOR_BFLOAT16:
+            return index_factory.HasIndex<bf16>(index_name);
+        case VecType::VECTOR_FLOAT16:
+            return index_factory.HasIndex<fp16>(index_name);
+        case VecType::VECTOR_SPARSE_FLOAT:
+            if (index_name != IndexEnum::INDEX_SPARSE_INVERTED_INDEX && index_name != IndexEnum::INDEX_SPARSE_WAND) {
+                return false;
```
Comment (foxspy) [inline include/knowhere/comp/knowhere_check.h:35]: should not limit INDEX_SPARSE_INVERTED_INDEX/INDEX_SPARSE_WAND

File: include/knowhere/comp/knowhere_check.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+#include <string>
+
+#include "knowhere/comp/index_param.h"
+#include "knowhere/index/index_factory.h"
+namespace knowhere {
+namespace KnowhereCheck {
+bool
+IndexTypeAndDataTypeCheck(const std::string& index_name, VecType data_type) {
+    auto& index_factory = IndexFactory::Instance();
```
Comment (chasingegg) [inline include/knowhere/comp/knowhere_check.h:23]: can we also check metric type when we create index?
we haven't make metric type as a template param through

**Pull request merged**